src/library/scala/collection/Iterable.scala

```diff
@@ -597,11 +597,13 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
       val bldr = m.getOrElseUpdate(k, iterableFactory.newBuilder[B])
       bldr += f(elem)
     }
-    var result = immutable.Map.empty[K, CC[B]]
-    m.foreach { case (k, v) =>
-      result = result + ((k, v.result()))
+    object result extends Function[(K, Builder[B, CC[B]]), Unit] {
```

Comment:
use `class` and `AbstractFunction` ? 🤔 

- https://github.com/scala/scala/pull/9216

Reply:
Thanks, I will also add a lint for that.